datatypes/src/raster/arrow_conversion.rs

```diff
@@ -0,0 +1,302 @@
+use super::{Grid2D, GridOrEmpty, GridOrEmpty2D, GridSize, Pixel, RasterTile2D, TypedGrid2D};
+use crate::{
+    primitives::SpatialPartitioned, raster::RasterDataType,
+    spatial_reference::SpatialReferenceOption, util::Result,
+};
+use arrow::{
+    array::{Array, ArrayRef, PrimitiveBuilder},
+    datatypes::{
+        Field, Float32Type, Float64Type, Int16Type, Int32Type, Int64Type, Int8Type, Schema,
+        UInt16Type, UInt32Type, UInt64Type, UInt8Type,
+    },
+    ipc::{
+        writer::{FileWriter, IpcWriteOptions},
+        CompressionType,
+    },
+    record_batch::RecordBatch,
+};
+use std::{collections::HashMap, sync::Arc};
+
+pub const RASTER_DATA_FIELD_NAME: &str = "data";
+pub const SPATIAL_PARTITION_KEY: &str = "spatialPartition";
+pub const X_SIZE_KEY: &str = "xSize";
+pub const Y_SIZE_KEY: &str = "ySize";
+pub const TIME_KEY: &str = "time";
+pub const SPATIAL_REF_KEY: &str = "spatialReference";
+
+pub fn raster_tile_2d_to_arrow_ipc_file<P: Pixel>(
+    tile: RasterTile2D<P>,
+    spatial_ref: SpatialReferenceOption,
+) -> Result<Vec<u8>> {
+    let record_batch = raster_tile_2d_to_arrow_record_batch(tile, spatial_ref)?;
+
+    let mut file_writer = FileWriter::try_new_with_options(
+        Vec::new(),
+        &record_batch.schema(),
+        IpcWriteOptions::default().try_with_compression(Some(CompressionType::LZ4_FRAME))?,
+    )?;
+    file_writer.write(&record_batch)?;
+    file_writer.finish()?;
+
+    Ok(file_writer.into_inner()?)
+}
+
+fn raster_tile_2d_to_arrow_record_batch<P: Pixel>(
+    tile: RasterTile2D<P>,
+    spatial_ref: SpatialReferenceOption,
+) -> Result<RecordBatch> {
+    let spatial_partition = tile.tile_information().spatial_partition();
+    let metadata: HashMap<String, String> = [
+        (
+            SPATIAL_PARTITION_KEY.to_string(),
```

Comment:
i think we should send a GeoTransform since we send the shape of the array anyway. This will also avoid some math in python.

Reply:
replace bbox with geo transform

Reply:
[ed9cb0d](https://github.com/geo-engine/geoengine/pull/737/commits/ed9cb0d63548df14ce92259099211ec63d550e2f)